// typeor/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.strOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = ""; }
    return typeof v === "string" ? v : defaultValue;
};
exports.numOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = 0; }
    return typeof v === "number" ? v : defaultValue;
};
exports.numHardOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = 0; }
    return typeof v === "number" && !isNaN(v) ? v : defaultValue;
};
exports.boolOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = false; }
    return typeof v === "boolean" ? v : defaultValue;
};
exports.objOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = {}; }
    return typeof v === "object" && !Array.isArray(v) && v !== null ? v : defaultValue;
};
exports.objNullOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = {}; }
    return typeof v === "object" && !Array.isArray(v) ? v : defaultValue;
};
exports.funcOr = function (v, defaultValue) {
    return typeof v === "function" ? v : defaultValue;
};
exports.arrayOr = function (v, defaultValue) {
    if (defaultValue === void 0) { defaultValue = []; }
    return Array.isArray(v) ? v : defaultValue;
};
